test/modern_bpf/test_suites/syscall_enter_suite/eventfd_e.cpp

```diff
@@ -0,0 +1,46 @@
+#include "../../event_class/event_class.h"
+
+#if defined(__NR_eventfd) && defined(__NR_close)
+TEST(SyscallEnter, eventfdE)
+{
+	auto evt_test = new event_test(__NR_eventfd, ENTER_EVENT);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	uint32_t initval = 3;
+	int flags = 0;
+	int32_t fd = syscall(__NR_eventfd, initval, flags);
```

Comment:
The `eventfd` and `eventfd2` syscalls seem to be interesting. Actually based on the [man page of eventfd](https://man7.org/linux/man-pages/man2/eventfd.2.html), it states

> There are two underlying Linux system calls: eventfd() and the more recent eventfd2().  The former system call does not implement a flags argument.  The latter system call implements the flags values described above.  The glibc wrapper function will use eventfd2() where it is available.

The Linux kernel source code seems to confirm that [here](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/fs/eventfd.c#n454).

So for the `eventfd`, the flags argument should not be necessary... only for the `eventfd2` syscall.

Was the scap event then actually implemented with `eventfd2` in mind?

Reply:
Yeah, that is the same case of `inotify_init` and `inotify_init1`, `inotify_init` doesn't have the flag but we catch it anyway. The solution could be to create 2 separate events as I have already tracked it in https://github.com/falcosecurity/libs/issues/515 :thinking:

Reply:
Ok.. will look at them as well... but still I wonder passing flags to the syscall even if the syscall does not use it. And agree, I would rather split those events into 2 distinct ones.

Reply:
During this refactor we need to understand which params we need because I agree with you that some of these params are not used at all!!